develop/Screens/ViewModels/ListViewModel.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  ListViewModel.swift
+//  TodoList
+//
+//  Created by Alexander Volokhov
+//
+
+import Foundation
+
+class ListViewModel: ObservableObject {
+    
+    @Published var items: [ItemModel] = [] {
+        didSet {
+            saveItems()
+        }
+    }
+    
+    let itemsKey: String = "items_list"
+    
+    init() {
+        getItems()
+    }
+    
+    func getItems() {
+        guard
+            let data = UserDefaults.standard.data(forKey: itemsKey),
+            let savedItems = try? JSONDecoder().decode([ItemModel].self, from: data)
+        else { return }
+
+        self.items = savedItems
+    }
+    
+    func deleteItem(indexSet: IndexSet) {
+        items.remove(atOffsets: indexSet)
+    }
+    
+    func moveItem(from: IndexSet, to: Int) {
+        items.move(fromOffsets: from, toOffset: to)
+    }
+    
+    func addItem(title: String) {
+        let newItem = ItemModel(title: title, isCompleted: false)
+        items.append(newItem)
+    }
+    
+    func updateItem(item: ItemModel) {
+        if let index = items.firstIndex(where: { $0.id == item.id }) {
+            items[index] = item.updateCompletion()
+        }
+    }
+    
+    func saveItems() {
+        if let encodedData = try? JSONEncoder().encode(items) {
```

Comment:
error case not handled

ANd for future:
If you have if let { } and no other code: use guard instead of if

Reply:
47 line the same issue